api/solarperformanceinsight_api/pvmodeling.py

```diff
@@ -0,0 +1,95 @@
+from pvlib.tracking import SingleAxisTracker  # type: ignore
+
+from . import models
+
+
+def construct_location(system: models.PVSystem) -> Location:
+    """Construct a pvlib Location object from a PVSystem"""
+    return Location(
+        latitude=system.latitude,
+        longitude=system.longitude,
+        altitude=system.elevation,
+        name=system.name,
+    )
+
+
+def construct_tracker(
+    tracking: Union[models.FixedTracking, models.SingleAxisTracking]
+) -> Tuple[Union[PVSystem, SingleAxisTracker], Dict[str, Any]]:
+    """Return the appropriate pvlib.pvsystem class and tracking keywords
+    depending on the type of tracking unit"""
+    if isinstance(tracking, models.FixedTracking):
+        return PVSystem, dict(
+            surface_tilt=tracking.tilt, surface_azimuth=tracking.azimuth
+        )
+    elif isinstance(tracking, models.SingleAxisTracking):
+        return SingleAxisTracker, dict(
+            axis_tilt=tracking.axis_tilt,
+            axis_azimuth=tracking.axis_azimuth,
+            gcr=tracking.gcr,
+            backtrack=tracking.backtracking,
+        )
+    else:
+        raise TypeError("Only FixedTracking and SingleAxisTracking currently supported")
+
+
+def construct_pvsystem(
+    inverter: models.Inverter, albedo: float
+) -> Union[PVSystem, SingleAxisTracker]:
+    """Construct a pvlib.pvsystem.PVSystem (or SingleAxisTracker) from an
+    Inverter object"""
+    # only a single array for now until pvlib#1076
+    # otherwise one pvsystem/pvarray per array
+    array = inverter.arrays[0]
+    common_kwargs = dict(
+        albedo=albedo,
+        module=array.make_model,
+        module_parameters=array.module_parameters.dict(),
+        temperature_model_parameters=array.temperature_model_parameters.dict(),
+        modules_per_string=array.modules_per_string,
+        strings_per_inverter=array.strings,
+        inverter=inverter.make_model,
+        inverter_parameters=inverter.inverter_parameters.dict(),
+        name=inverter.name,
+    )
+    if inverter.losses is not None:
+        common_kwargs["losses_parameters"] = inverter.losses.dict()
+    system_model, tracking_kwargs = construct_tracker(array.tracking)
+    return system_model(**tracking_kwargs, **common_kwargs)
+
+
+def extract_pvlib_models(inverter: models.Inverter) -> Dict[str, str]:
+    """Extract the pvlib.modelchain.ModelChain _model keywords from an
+    Inverter definition"""
+    models = dict(inverter._pvlib_models)
+    models.update(
+        {
+            k: v
+            for k, v in inverter.dict().items()
+            if k.endswith("_model") and k != "make_model"
+        }
+    )
+    return models
+
+
+def construct_modelchains(system: models.PVSystem) -> List[ModelChain]:
+    """Construct a pvlib.modelchain.ModelChain object for each Inverter
+    in system"""
+    location = construct_location(system=system)
+    out = []
+    for i, inverter in enumerate(system.inverters):
```

Comment:
why the enumeration? `i` is not used.

Reply:
right, I was getting ahead of myself there thinking about descriptive error messages (w/ location of problematic inverter) if the ModelChain construction failed. I don't think that failure is currently possible with our strict schema until maybe multiple arrays are implemented